as/cNFT/assembly/models/persistent_tokens.ts

```diff
@@ -113,6 +113,50 @@ export class PersistentTokens {
         return token
     }
 
+    /**
+     * @param tokenId: Id of the token
+     * @param bidderId: accountId of the bidder
+    */
+
+    cnft_transfer(tokenId: TokenId, bidderId: AccountId): void {
+
+        /* Getting stored token from tokenId */
+
+        const token = this._tmap.getSome(tokenId)
+        const accountId = token.owner_id
+
+        /* Setting new details of the token */
+
+        token.prev_owner_id = token.owner_id
+        token.owner_id = bidderId
+
+        this._tmap.set(tokenId, token)
+
+
+        /* Deleting token from previous owner */
+
+        this._deleteFromAccountTokenSet(tokenId, accountId);
+
+
+        /* Storing token with the new owner's accountId */
```

Comment:
Same for the logic below this line,
You can just do `this.add(tokenId, token, accountId)` with the new info

Reply:
Yeah. Sure :-)